src/CachePeer.cc

```diff
@@ -114,6 +118,209 @@ CachePeer::connectTimeout() const
+{
+    if (options.proxy_only)
+        os << " proxy-only";
+
+    if (options.no_query)
+        os << " no-query";
+
+    if (options.background_ping)
+        os << " background-ping";
+
+    if (options.no_digest)
+        os << " no-digest";
+
+    if (options.default_parent)
+        os << " default";
+
+    if (options.roundrobin)
+        os << " round-robin";
+
+    if (options.carp)
+        os << " carp";
+
+#if USE_AUTH
+    if (options.userhash)
+        os << " userhash";
+#endif
+
+    if (options.sourcehash)
+        os << " sourcehash";
+
+    if (options.weighted_roundrobin)
+        os << " weighted-round-robin";
+
+    if (options.mcast_responder)
+        os << " multicast-responder";
+
+#if PEER_MULTICAST_SIBLINGS
+    if (options.mcast_siblings)
+        os << " multicast-siblings";
+#endif
+
+    if (weight != 1)
+        os << " weight=" << weight;
+
+    if (options.closest_only)
+        os << " closest-only";
+
+#if USE_HTCP
+    if (options.htcp)
+    {
+        std::vector<const char *> opts {};
+        if (options.htcp_oldsquid)
+            opts.push_back("oldsquid");
+        if (options.htcp_no_clr)
+            opts.push_back("no-clr");
+        if (options.htcp_no_purge_clr)
+            opts.push_back("no-purge-clr");
+        if (options.htcp_only_clr)
+            opts.push_back("only-clr");
+        os << AsList(opts).prefixedBy(" htcp=").delimitedBy(",");
+    }
+#endif
+
+    if (options.no_netdb_exchange)
+        os << " no-netdb-exchange";
+
+#if USE_DELAY_POOLS
+    if (options.no_delay)
+        os << " no-delay";
+#endif
+
+    if (login)
+        os << " login=" << login;
+
+    if (mcast.ttl > 0)
+        os << " ttl=" << mcast.ttl;
+
+    if (connect_timeout_raw > 0)
+        os << " connect-timeout=" << connect_timeout_raw;
+
+    if (connect_fail_limit != PEER_TCP_MAGIC_COUNT)
+        os << " connect-fail-limit=" << connect_fail_limit;
+
+#if USE_CACHE_DIGESTS
+    if (digest_url)
+        os << " digest-url=" << digest_url;
+#endif
+
+    if (options.allow_miss)
+        os << " allow-miss";
+
+    if (options.no_tproxy)
+        os << " no-tproxy";
+
+    if (max_conn > 0)
+        os << " max-conn=" << max_conn;
+
+    if (standby.limit > 0)
+        os << " standby=" << standby.limit;
+
+    if (options.originserver)
+        os << " originserver";
+
+    if (domain)
+        os << " forceddomain=" << domain;
+
+    if (connection_auth == 0)
+        os << " connection-auth=off";
+    else if (connection_auth == 1)
+        os << " connection-auth=on";
+    // else default connection-auth=auto
+
+    secure.dumpCfg(os, "tls-");
+}
+
+void
+CachePeer::reportStatistics (std::ostream& yaml)
+{
+    yaml <<
+         spaces(2) << "- name: " << name << "\n" <<
+         spaces(4) << "type: " << typeString() << '\n' <<
+         spaces(4) << "http address: " << host << ':' << http_port << '\n';
+    if (icp.port)
+        yaml << spaces(4) << "icp address:" << host << ':' << icp.port << '\n';
```

Comment:
I think we ought to use all-caps for these protocol name abbreviations. This is how the standards and the vast majority of folks spell them. Why be different in this for-human output that does not have readability, clashes, and other concerns that C++ names have?

```suggestion
         spaces(4) << "HTTP address: " << host << ':' << http_port << '\n';
    if (icp.port)
        yaml << spaces(4) << "ICP address:" << host << ':' << icp.port << '\n';
```

Reply:
Mostly for consistency, and with machine parsers in mind. Personal preference comes into play as well.
I think it's a nit, I'll change it if you feel strongly about it

Reply:
> Alex: Why be different in this for-human output that does not have readability, clashes, and other concerns that C++ names have?

> Francesco: Mostly for consistency

The proposed format (change) is not consistent with current Squid output and, more importantly, is very inconsistent with subject area norms. That problem trumps any desire to make for-admin output text consistent with for-developers source code names (if that is what you are implying). Not to mention that there are plenty of Squid source code names where HTTP is (correctly) formatted as `Http`, `http`, or (incorrectly) `HTTP`, so no single variant will be truly consistent with C++ code names.

> and with machine parsers in mind

Why would a machine parser care whether the token it is parsing contains "HTTP" or "http"?!

> I'll change it if you feel strongly about it

I hope this decision can be made based on (objective) merits rather than the (subjective) strength of our convictions.

Reply:
> I hope this decision can be made based on (objective) merits rather than the (subjective) strength of our convictions.

I don't think there's a measurable benefit by taking either choice, so I'm not sure we could converge on some objective merit. But that's not a big deal, I have no attachment to either choice so I'm changing it